### PR TITLE
Bibles can convert other bibles

### LINKED ISSUE #72884: Bibles can no longer convert other bibles
Reporting client version: 514.1589

## Round ID:
198697

## Testmerges:
- [Refactors crew records](https://www.github.com/tgstation/tgstation/pull/72725)

## Reproduction:
Be chaplain
Click on another bible with your bible
[Your bible] is too big for the bible!

Tried every combination of:
Right clicks
Left clicks
Combat mode on
Combat mode off
Having something inside my bible
Having something inside the target bible

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
Fixes #72884
## Changelog
:cl:
fix: Bibles can convert other bibles
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
why arent we calling parent anymore

**Comment by SyncIt21** [comment]
> why arent we calling parent anymore

Because it doesn't do anything?

**Review by MrMelbert — CHANGES_REQUESTED** [review]
None of the uses of `COMSIG_BIBLE_SMACKED` were updated

Regardless, I think the better fix for this is to implement `attackby_storage_insert` -
```
/obj/item/storage/book/bible/attackby_storage_insert(datum/storage, atom/storage_holder, mob/user)
    return !istype(storage_holder, /obj/item/storage/book/bible)
```
Instead of needing to futz with changing around the bible's attack chain and handing the repercussions of that, just prevent bibles from inserting into bibles. 

I BELIEVE this works, I did this fix for a branch which I haven't submitted yet

**Comment by SyncIt21** [comment]
> None of the uses of `COMSIG_BIBLE_SMACKED` were updated
> 
> Regardless, I think the better fix for this is to implement `attackby_storage_insert` -
> 
> ```
> /obj/item/storage/book/bible/attackby_storage_insert(datum/storage, atom/storage_holder, mob/user)
>     return !istype(storage_holder, /obj/item/storage/book/bible)
> ```
> 
> Instead of needing to futz with changing around the bible's attack chain and handing the repercussions of that, just prevent bibles from inserting into bibles.
> 
> I BELIEVE this works, I did this fix for a branch which I haven't submitted yet

I'll test it out

**Comment by SyncIt21** [comment]
It worked

File: code/game/objects/items/storage/book.dm [hunk 1]
```diff
@@ -186,12 +186,13 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "burning",
...
 
+/obj/item/storage/book/bible/attackby_storage_insert(datum/storage, atom/storage_holder, mob/user)
+	return !istype(storage_holder, /obj/item/storage/book/bible)
+
 /obj/item/storage/book/bible/afterattack(atom/bible_smacked, mob/user, proximity)
 	. = ..()
 	if(!proximity)
 		return
-	if(SEND_SIGNAL(bible_smacked, COMSIG_BIBLE_SMACKED, user, proximity) & COMSIG_END_BIBLE_CHAIN)
-		return . | AFTERATTACK_PROCESSED_ITEM
```
Comment (JohnFulpWillard) [inline code/game/objects/items/storage/book.dm:194]: Why're we removing this entirely?
If it's not needed, we should delete the defines too
But that would also mean limiting downstream use of it

Comment (SyncIt21) [reply code/game/objects/items/storage/book.dm:194]: I'll put it back

**Comment by SyncIt21** [comment]
Merge/Review Anyone?

**Pull request merged**